packages/core/types/@chainlink/index.d.ts

```diff
@@ -130,6 +131,18 @@ declare module '@chainlink/types' {
     rpcPort?: number
   }
 
+  export type EnvDefaults = { [name: string]: string }
+
+  /**
+   * To add a default override for an env var not present here,
+   * ensure that the adapter context from `expose` is passed
+   * to the `getEnv` call for the variable in question. See
+   * `WS_ENABLED` for example usage.
+   */
+  export type EnvDefaultOverrides = {
+    WS_ENABLED?: 'true' | 'false'
```

Comment:
What if we held these as their proper type (e.g. `boolean` here) and then `String(envDefaultOverride)` them later?

Might be more intuitive for developers.

Reply:
Actually I see why this is a problem, we `getEnv` would need to be heavily reworked with a schema of sorts. 
```
{
CACHE_MAX_ITEMS: {
default: 1000,
type: number
}
}
```
Then we would have helper functions to convert the environment variable string to a number.
So if it comes from env var use the `number` helper, otherwise use the default which is already a number.

Reply:
Yeah, I like the concept but I would like to avoid that for this initial implementation. The defaults act as if they were any other env var pulled in, which is always a string if present.